<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from /home/mnaveau/devel/workspace/src/catkin/robots/robot_properties/robot_properties_quadruped/xacro/quadruped_old.urdf.xacro | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="quadruped" xmlns:controller="http://playerstage.sourceforge.net/gazebo/xmlschema/#controller" xmlns:interface="http://playerstage.sourceforge.net/gazebo/xmlschema/#interface" xmlns:sensor="http://playerstage.sourceforge.net/gazebo/xmlschema/#sensor" xmlns:xacro="http://playerstage.sourceforge.net/gazebo/xmlschema/#interface">
  <link name="base_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="2.0"/>
      <inertia ixx="0.00708333333333" ixy="0" ixz="0" iyy="0.0270833333333" iyz="0" izz="0.0333333333333"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robot_properties_quadruped/meshes/cube.obj" scale="0.4 0.2 0.05"/>
      </geometry>
      <material name="white">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.4 0.2 0.05"/>
      </geometry>
    </collision>
  </link>
  <!-- <xacro:leg prefix="FL" rx="${0.}" ry="${0.}"/> -->
  <joint name="FL_HFE" type="revolute">
    <parent link="base_link"/>
    <child link="FL_upperleg"/>
    <limit effort="1000" lower="-10" upper="10" velocity="1000"/>
    <axis xyz="0 1 0"/>
    <origin rpy="0 0 0" xyz="0.2 0.1 0"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="FL_upperleg">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.08"/>
      <geometry>
        <!-- <cylinder length="${upper_leg_length}" radius="${upper_leg_radius}"/> -->
        <mesh filename="package://robot_properties_quadruped/meshes/cylinder.obj" scale="0.025 0.025 0.16"/>
      </geometry>
      <material name="grey">
        <color rgba=".4 .4 .4 .1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.08"/>
      <geometry>
        <cylinder length="0.16" radius="0.025"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 -0.0793"/>
      <mass value="0.162"/>
      <inertia ixx="0.000469609" ixy="0.0" ixz="0.0" iyy="0.000469609" iyz="0.0" izz="1e-6"/>
    </inertial>
  </link>
  <joint name="FL_KFE" type="revolute">
    <parent link="FL_upperleg"/>
    <child link="FL_shank"/>
    <limit effort="1000" velocity="10000"/>
    <axis xyz="0 1 0"/>
    <origin rpy="0 0 0" xyz="0 0 -0.16"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="FL_shank">
    <contact>
      <lateral_friction value="1.0"/>
      <restitution value="0.5"/>
    </contact>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.08"/>
      <geometry>
        <!-- <cylinder length="${upper_leg_length}" radius="${upper_leg_radius}"/> -->
        <mesh filename="package://robot_properties_quadruped/meshes/cylinder.obj" scale="0.025 0.025 0.16"/>
      </geometry>
      <material name="grey">
        <color rgba=".4 .4 .4 .1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.08"/>
      <geometry>
        <cylinder length="0.16" radius="0.025"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 -0.0756"/>
      <mass value="0.021"/>
      <inertia ixx="0.000087575" ixy="0.0" ixz="0.0" iyy="0.000087575" iyz="0.0" izz="1e-6"/>
    </inertial>
  </link>
  <joint name="FL_END" type="fixed">
    <parent link="FL_shank"/>
    <child link="FL_contact"/>
    <origin rpy="0 0 0" xyz="0 0 -0.16"/>
  </joint>
  <link name="FL_contact">
    <contact>
      <lateral_friction value="1.0"/>
      <restitution value="0.5"/>
    </contact>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <!-- <sphere radius="${upper_leg_radius}"/> -->
        <mesh filename="package://robot_properties_quadruped/meshes/sphere.obj" scale="0.025 0.025 0.025"/>
      </geometry>
      <material name="red">
        <color rgba="1. 0. 0. 1."/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.025"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.01"/>
      <inertia ixx="0." ixy="0.0" ixz="0.0" iyy="0." iyz="0.0" izz="0."/>
    </inertial>
  </link>
  <joint name="FR_HFE" type="revolute">
    <parent link="base_link"/>
    <child link="FR_upperleg"/>
    <limit effort="1000" lower="-10" upper="10" velocity="1000"/>
    <axis xyz="0 1 0"/>
    <origin rpy="0 0 0" xyz="0.2 -0.1 0"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="FR_upperleg">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.08"/>
      <geometry>
        <!-- <cylinder length="${upper_leg_length}" radius="${upper_leg_radius}"/> -->
        <mesh filename="package://robot_properties_quadruped/meshes/cylinder.obj" scale="0.025 0.025 0.16"/>
      </geometry>
      <material name="grey">
        <color rgba=".4 .4 .4 .1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.08"/>
      <geometry>
        <cylinder length="0.16" radius="0.025"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 -0.0793"/>
      <mass value="0.162"/>
      <inertia ixx="0.000469609" ixy="0.0" ixz="0.0" iyy="0.000469609" iyz="0.0" izz="1e-6"/>
    </inertial>
  </link>
  <joint name="FR_KFE" type="revolute">
    <parent link="FR_upperleg"/>
    <child link="FR_shank"/>
    <limit effort="1000" velocity="10000"/>
    <axis xyz="0 1 0"/>
    <origin rpy="0 0 0" xyz="0 0 -0.16"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="FR_shank">
    <contact>
      <lateral_friction value="1.0"/>
      <restitution value="0.5"/>
    </contact>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.08"/>
      <geometry>
        <!-- <cylinder length="${upper_leg_length}" radius="${upper_leg_radius}"/> -->
        <mesh filename="package://robot_properties_quadruped/meshes/cylinder.obj" scale="0.025 0.025 0.16"/>
      </geometry>
      <material name="grey">
        <color rgba=".4 .4 .4 .1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.08"/>
      <geometry>
        <cylinder length="0.16" radius="0.025"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 -0.0756"/>
      <mass value="0.021"/>
      <inertia ixx="0.000087575" ixy="0.0" ixz="0.0" iyy="0.000087575" iyz="0.0" izz="1e-6"/>
    </inertial>
  </link>
  <joint name="FR_END" type="fixed">
    <parent link="FR_shank"/>
    <child link="FR_contact"/>
    <origin rpy="0 0 0" xyz="0 0 -0.16"/>
  </joint>
  <link name="FR_contact">
    <contact>
      <lateral_friction value="1.0"/>
      <restitution value="0.5"/>
    </contact>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <!-- <sphere radius="${upper_leg_radius}"/> -->
        <mesh filename="package://robot_properties_quadruped/meshes/sphere.obj" scale="0.025 0.025 0.025"/>
      </geometry>
      <material name="red">
        <color rgba="1. 0. 0. 1."/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.025"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.01"/>
      <inertia ixx="0." ixy="0.0" ixz="0.0" iyy="0." iyz="0.0" izz="0."/>
    </inertial>
  </link>
  <joint name="HL_HFE" type="revolute">
    <parent link="base_link"/>
    <child link="HL_upperleg"/>
    <limit effort="1000" lower="-10" upper="10" velocity="1000"/>
    <axis xyz="0 1 0"/>
    <origin rpy="0 0 0" xyz="-0.2 0.1 0"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="HL_upperleg">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.08"/>
      <geometry>
        <!-- <cylinder length="${upper_leg_length}" radius="${upper_leg_radius}"/> -->
        <mesh filename="package://robot_properties_quadruped/meshes/cylinder.obj" scale="0.025 0.025 0.16"/>
      </geometry>
      <material name="grey">
        <color rgba=".4 .4 .4 .1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.08"/>
      <geometry>
        <cylinder length="0.16" radius="0.025"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 -0.0793"/>
      <mass value="0.162"/>
      <inertia ixx="0.000469609" ixy="0.0" ixz="0.0" iyy="0.000469609" iyz="0.0" izz="1e-6"/>
    </inertial>
  </link>
  <joint name="HL_KFE" type="revolute">
    <parent link="HL_upperleg"/>
    <child link="HL_shank"/>
    <limit effort="1000" velocity="10000"/>
    <axis xyz="0 1 0"/>
    <origin rpy="0 0 0" xyz="0 0 -0.16"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="HL_shank">
    <contact>
      <lateral_friction value="1.0"/>
      <restitution value="0.5"/>
    </contact>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.08"/>
      <geometry>
        <!-- <cylinder length="${upper_leg_length}" radius="${upper_leg_radius}"/> -->
        <mesh filename="package://robot_properties_quadruped/meshes/cylinder.obj" scale="0.025 0.025 0.16"/>
      </geometry>
      <material name="grey">
        <color rgba=".4 .4 .4 .1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.08"/>
      <geometry>
        <cylinder length="0.16" radius="0.025"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 -0.0756"/>
      <mass value="0.021"/>
      <inertia ixx="0.000087575" ixy="0.0" ixz="0.0" iyy="0.000087575" iyz="0.0" izz="1e-6"/>
    </inertial>
  </link>
  <joint name="HL_END" type="fixed">
    <parent link="HL_shank"/>
    <child link="HL_contact"/>
    <origin rpy="0 0 0" xyz="0 0 -0.16"/>
  </joint>
  <link name="HL_contact">
    <contact>
      <lateral_friction value="1.0"/>
      <restitution value="0.5"/>
    </contact>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <!-- <sphere radius="${upper_leg_radius}"/> -->
        <mesh filename="package://robot_properties_quadruped/meshes/sphere.obj" scale="0.025 0.025 0.025"/>
      </geometry>
      <material name="red">
        <color rgba="1. 0. 0. 1."/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.025"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.01"/>
      <inertia ixx="0." ixy="0.0" ixz="0.0" iyy="0." iyz="0.0" izz="0."/>
    </inertial>
  </link>
  <joint name="HR_HFE" type="revolute">
    <parent link="base_link"/>
    <child link="HR_upperleg"/>
    <limit effort="1000" lower="-10" upper="10" velocity="1000"/>
    <axis xyz="0 1 0"/>
    <origin rpy="0 0 0" xyz="-0.2 -0.1 0"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="HR_upperleg">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.08"/>
      <geometry>
        <!-- <cylinder length="${upper_leg_length}" radius="${upper_leg_radius}"/> -->
        <mesh filename="package://robot_properties_quadruped/meshes/cylinder.obj" scale="0.025 0.025 0.16"/>
      </geometry>
      <material name="grey">
        <color rgba=".4 .4 .4 .1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.08"/>
      <geometry>
        <cylinder length="0.16" radius="0.025"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 -0.0793"/>
      <mass value="0.162"/>
      <inertia ixx="0.000469609" ixy="0.0" ixz="0.0" iyy="0.000469609" iyz="0.0" izz="1e-6"/>
    </inertial>
  </link>
  <joint name="HR_KFE" type="revolute">
    <parent link="HR_upperleg"/>
    <child link="HR_shank"/>
    <limit effort="1000" velocity="10000"/>
    <axis xyz="0 1 0"/>
    <origin rpy="0 0 0" xyz="0 0 -0.16"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="HR_shank">
    <contact>
      <lateral_friction value="1.0"/>
      <restitution value="0.5"/>
    </contact>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.08"/>
      <geometry>
        <!-- <cylinder length="${upper_leg_length}" radius="${upper_leg_radius}"/> -->
        <mesh filename="package://robot_properties_quadruped/meshes/cylinder.obj" scale="0.025 0.025 0.16"/>
      </geometry>
      <material name="grey">
        <color rgba=".4 .4 .4 .1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.08"/>
      <geometry>
        <cylinder length="0.16" radius="0.025"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 -0.0756"/>
      <mass value="0.021"/>
      <inertia ixx="0.000087575" ixy="0.0" ixz="0.0" iyy="0.000087575" iyz="0.0" izz="1e-6"/>
    </inertial>
  </link>
  <joint name="HR_END" type="fixed">
    <parent link="HR_shank"/>
    <child link="HR_contact"/>
    <origin rpy="0 0 0" xyz="0 0 -0.16"/>
  </joint>
  <link name="HR_contact">
    <contact>
      <lateral_friction value="1.0"/>
      <restitution value="0.5"/>
    </contact>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <!-- <sphere radius="${upper_leg_radius}"/> -->
        <mesh filename="package://robot_properties_quadruped/meshes/sphere.obj" scale="0.025 0.025 0.025"/>
      </geometry>
      <material name="red">
        <color rgba="1. 0. 0. 1."/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.025"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.01"/>
      <inertia ixx="0." ixy="0.0" ixz="0.0" iyy="0." iyz="0.0" izz="0."/>
    </inertial>
  </link>
</robot>
